eth_defi/gmx/core/markets.py

```diff
@@ -0,0 +1,330 @@
+"""
+GMX Markets Data Module
+
+This module provides access to GMX protocol market information and trading pairs.
+"""
+
+import logging
+from dataclasses import dataclass
+from typing import Optional, Any
+
+from eth_typing import HexAddress
+
+from cchecksum import to_checksum_address
+
+from eth_defi.gmx.config import GMXConfig
+from eth_defi.gmx.contracts import get_contract_addresses, get_reader_contract, get_tokens_address_dict
+from eth_defi.gmx.core.oracle import OraclePrices
+from eth_defi.gmx.types import MarketSymbol, MarketData
+
+
+@dataclass
```

Comment:
You can use `@dataclass(slots=True)` if there are no dynamic attributes inserted on the class.

This will also cause edit time and run-time error if you accidentally try to set a variable that does not exist. 

Slots objects are also 20% faster.